auth/src/main/java/com/firebase/ui/auth/ui/AcquireEmailHelper.java

```diff
@@ -80,30 +80,22 @@ private void startEmailHandler(String email, List<String> providers) {
             mActivityHelper.startActivityForResult(registerIntent, RC_REGISTER_ACCOUNT);
         } else {
             // account does exist
-            for (String provider : providers) {
-                if (provider.equalsIgnoreCase(EmailAuthProvider.PROVIDER_ID)) {
-                    Intent signInIntent = SignInActivity.createIntent(
-                            mActivityHelper.getApplicationContext(),
-                            mActivityHelper.getFlowParams(),
-                            email);
-                    mActivityHelper.startActivityForResult(signInIntent, RC_SIGN_IN);
-                    return;
-                }
-
-                Intent intent = WelcomeBackIdpPrompt.createIntent(
+            if (providers.get(0).equalsIgnoreCase(EmailAuthProvider.PROVIDER_ID)) {
+                Intent signInIntent = SignInActivity.createIntent(
                         mActivityHelper.getApplicationContext(),
                         mActivityHelper.getFlowParams(),
-                        provider,
-                        null,
                         email);
-                mActivityHelper.startActivityForResult(intent, RC_WELCOME_BACK_IDP);
+                mActivityHelper.startActivityForResult(signInIntent, RC_SIGN_IN);
                 return;
             }
 
-            Intent signInIntent = new Intent(
-                    mActivityHelper.getApplicationContext(), SignInActivity.class);
-            signInIntent.putExtra(ExtraConstants.EXTRA_EMAIL, email);
-            mActivityHelper.startActivityForResult(signInIntent, RC_SIGN_IN);
+            Intent intent = WelcomeBackIdpPrompt.createIntent(
```

Comment:
This seems to be a behavior change from `SignInActivity` to `WelcomeBackIDPPrompt`.